RedHatInsights/compliance-frontend
# feat(PolicyDetails): RHINENG-10399 - Migrate Policy Rules Tab to API v2

This migrates the Policy Details Rules Tab to use the REST API v2.
It also introduces the treeTable functionality, it can and should be enabled for testing, but is disabled by default, until RHINENG-13519 is resolved:

**How to test:**

1) Enable REST API v2 & AsyncTable
2) Open the Policy details rules tab
3) Verify the rules are shown for each minor OS profile and rules filter, paginate and sort correctly.
4) Enable the ruleTree by passing on the `ruleTree` returned by `data` in the `TailoringsTab` component and pass it on to the rules table.
5) Verify the tree expands correctly, and loads the rules and rule details (These might not be the correctly assigned rules yet due to API changes still needed)


## Secure Coding Practices Checklist GitHub Link
- https://github.com/RedHatInsights/secure-coding-checklist

## Secure Coding Checklist
